### ISSUE TITLE
How to change color of '$'

### ISSUE THREAD

**Issue by bharadwaj-raju** [caiogondim/bullet-train-oh-my-zsh-theme#144]
How do I change the color of the `$ ` prompt?

I've tried using color escape codes in the `BULLETTRAIN_PROMPT_CHAR` variable but then thde `$` becomes invisible.

`echo`ing the same text gives the `$` in the proper color.

**Comment by tunnckoCore** [comment]
:+1: it's not only the color, we can't set anything to `BULLETTRAIN_PROMPT_CHAR`, i tried to set custom text or another segment, nothing works. It is just just empty when set anything different than `$`.

**Comment by bharadwaj-raju** [comment]
@tunnckoCore Strange, custom text works for me:

```sh
if [ -n "${BULLETTRAIN_PROMPT_CHAR+1}" ]; then
    BULLETTRAIN_PROMPT_CHAR="λ"
fi

**Comment by tunnckoCore** [comment]
Hm. Maybe. I tried with direct assign `BULLETTRAIN_PROMPT_CHAR=" foo "`

**Comment by bharadwaj-raju** [comment]
@tunnckoCore Are you setting it in your `.zshrc`? I suppose you have to set it in `~/.oh-my-zsh/themes/bullet-train.zsh_theme`

**Comment by tunnckoCore** [comment]
@bharadwaj-raju the whole idea for that variables are to control themes out of theme files.

**Comment by bharadwaj-raju** [comment]
@tunnckoCore Hmm. This seems to do it *outside* the theme source:

```bash
source ~/.path/to/bullet-train.zsh-theme  # do not use the 'plugins' or 'ZSH_THEME' of oh-my-zsh

if [ -n "${BULLETTRAIN_PROMPT_CHAR+1}" ]; then
    BULLETTRAIN_PROMPT_CHAR="λ"
fi
```

**Comment by tunnckoCore** [comment]
Don't see sense and value on your example. oh-my-zsh is already loaded the things in anyway.

In anyway, i stopped using ohz and this theme already.

**Comment by bharadwaj-raju** [comment]
> i stopped using ohz

I've switched to prezto and get a fresh set of issues... time to change theme.

**Comment by unluisco** [comment]
I am trying to change the prompt, with this code in .zshrc
``` BULLETTRAIN_PROMPT_CHAR="»" ```
but all I have is an invisible prompt
what I am do it wrong?

**Comment by revolter** [comment]
@unluisco, Maybe the terminal app you're using can't render that character properly?

**Comment by bharadwaj-raju** [comment]
Yes @unluisco, try pasting it into the prompt (or anywhere, say `cat`) and see if it shows up.

**Comment by unluisco** [comment]
now the prompt work fine, thanks

**Comment by caiogondim** [comment]
Amazing!
Can we close the issue?

**Comment by revolter** [comment]
> I've switched to prezto and get a fresh set of issues... time to change theme.

It looks like it's invalid now, as @bharadwaj-raju, the original reporter moved on. And @unluisco's problem was different from this issue.

**Comment by ChristianVermeulen** [comment]
@unluisco what was your fix? My prompt char is not showing either, not even if I use plain text.....

**Comment by revolter** [comment]
@ChristianVermeulen, Could you post the part of your `.zshrc` that fiddles with Bullet Train?

**Comment by stevensonmt** [comment]
@ChristianVermeulen make sure you put the character inside parenthesis and double quotes like so: (">>").

Can someone tell me how to set the color of the prompt character though?  I can change it with %F, %B, %f, %b and so on, but I don't know how to set those values to what I want.

**Comment by dawikur** [comment]
Maybe we should introduce new variable, something like `BULLETTRAIN_PROMPT_COLOR` ?

**Comment by jakemayeux** [comment]
My prompt character always renders correctly but I still cannot change the color

Even when I do:
BULLETTRAIN_PROMPT_CHAR=("%{$fg[red]%}>>")

The prompt shows as just ">>" but it is still green

**Comment by revolter** [comment]
Show us your `.zshrc`

**Comment by stevensonmt** [comment]
I have to periodically reload .zshrc for some reason for the color to stick.  So in the terminal just do `. <pathto/.zshrc>`
